oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/BanyanDBAggregationQueryDAO.java

```diff
@@ -64,6 +63,13 @@ public List<SelectedRecord> sortMetrics(TopNCondition condition, String valueCol
 
         // BanyanDB server-side TopN support for metrics pre-aggregation.
         if (schema.getTopNSpec() != null) {
+            // check if additional conditions are registered
+            if (CollectionUtils.isNotEmpty(additionalConditions)) {
+                final Set<String> additionalConditionNames = additionalConditions.stream().map(KeyValue::getKey).collect(Collectors.toSet());
+                if (!ImmutableSet.copyOf(schema.getTopNSpec().getGroupByTagNames()).containsAll(additionalConditionNames)) {
+                    throw new IOException("[" + Strings.join(additionalConditionNames, ',') + "] conditions are not registered as groupBy tags");
```

Comment:
Two questions,
1. I think group-by pre-aggregation should be `equal` or `no condition`, rather than `contains`? I know that we have one condition, so, contains means extractly the same. But I am trying to be precise. 
2. When this pre-aggregation doesn't fit the current query, I think failing back to on-demand group-by still works. We don't have to through exception, right?

Reply:
You're right. Fixed